nixos/modules/services/web-apps/gerrit.nix

```diff
@@ -224,11 +236,11 @@ in
         LimitNOFILE = 4096;
         StandardInput = "socket";
         StandardOutput = "journal";
-        StateDirectory = "gerrit";
+        StateDirectory = lib.mkIf (cfg.stateDir == "/var/lib/gerrit") "gerrit";
```

Comment:
Maybe it would be easier to just bindmount the directories specified in the options over the equivalent systemd dirs? Something like

```nix
        BindPaths = lib.optionals (cfg.stateDir != "/var/lib/gerrit") [ "${cfg.stateDir}:/var/lib/gerrit" ]
        ++ lib.optionals (cfg.cacheDir != "/var/cache/gerrit") [ "${cfg.cacheDir}:/var/cache/gerrit" ];
```

And then the rest of the service can just assume those dirs to always be in their expected location?